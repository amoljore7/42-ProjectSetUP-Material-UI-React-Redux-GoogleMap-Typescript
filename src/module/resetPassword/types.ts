export const RESET_PASSWORD_LOADING = 'RESET_PASSWORD_LOADING';
export const RESET_PASSWORD = 'RESET_PASSWORD';
export const RESET_PASSWORD_SUCCESS = 'RESET_PASSWORD_SUCCESS';
export const RESET_PASSWORD_FAIL = 'RESET_PASSWORD_FAIL';

export interface Loading {
  type: typeof RESET_PASSWORD_LOADING;
}
export interface Reset_Password {
  type: typeof RESET_PASSWORD;
}

export interface LoginSuccess {
  type: typeof RESET_PASSWORD_SUCCESS;
  payload: SuccessPayloadType;
}

export type SuccessPayloadType = {
  email: string;
};

export interface LoginFail {
  type: typeof RESET_PASSWORD_FAIL;
  payload: string;
}

export type LoginDispatch = Loading | Reset_Password | LoginSuccess | LoginFail;
