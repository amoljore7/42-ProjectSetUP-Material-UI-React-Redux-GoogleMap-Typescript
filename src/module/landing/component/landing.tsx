import React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { Container, Button, CssBaseline, makeStyles, Typography } from '@material-ui/core';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme) => ({
  headingStyle: {
    font: 'normal normal bold 60px/80px Futura',
    letterSpacing: '0px',
    color: '#000000',
    opacity: 1,
    textAlign: 'center'
  },
  headings: {
    font: 'normal normal medium 20px/27px Futura',
    color: '#707070',
    textAlign: 'center'
  },
  headings2: {
    color: '#707070',
    font: 'normal normal bold 30px/40px Futura',
    marginTop: '40px',
    marginBottom: '10px',
    textAlign: 'center'
  },
  wrapperStyle: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: theme.spacing(10)
  },
  link: {
    fontSize: '20px'
  },
  signIn: {
    margin: theme.spacing(3, 0, 2)
  }
}));
const themeDark = createMuiTheme({
  palette: {
    background: {
      default: '#F5F5FA'
    }
  }
});

type Props = RouteComponentProps;

const LandingPage: React.FC<Props> = (props: Props) => {
  const clickHandler = () => {
    props.history.push('/login');
  };

  const classes = useStyles();
  return (
    <MuiThemeProvider theme={themeDark}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <div data-testid="landingWrapper" className={classes.wrapperStyle}>
          <Typography component="h1" variant="h5" className={classes.headingStyle}>
            Healthful
          </Typography>
          <Typography className={classes.headings2}>You are accessing PHI</Typography>
          <h5 className={classes.headings}>
            By clicking ’Sign in’ you agree that you will only use protected health information
            (PHI) to provide, coordinate or manage healthcare treatment for this patient
          </h5>
          <Button
            className={classes.signIn}
            style={{ backgroundColor: '#4491E0', color: '#FFFFFF' }}
            variant="contained"
            onClick={clickHandler}>
            Sign In
          </Button>
        </div>
      </Container>
    </MuiThemeProvider>
  );
};

export default LandingPage;
