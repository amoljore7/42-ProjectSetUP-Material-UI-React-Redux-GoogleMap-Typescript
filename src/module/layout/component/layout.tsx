import React from 'react';
import { Route, RouteComponentProps, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import Navbar from '../../../components/navbar';
import rootReducer from '../../../root-reducer';

import ResourceRouting from '../../resources/routing';
import PageNotFound from '../../pageNotFound/pageNotFound';
const Home = () => <h1>Home</h1>;
const Dashboard = () => <h1>Dashboard</h1>;
const Patients = () => <h1>Patients</h1>;
const Analytics = () => <h1>Analytics</h1>;

type Props = RouteComponentProps;
const Layout: React.FC<Props> = (props: Props) => {
  return (
    <React.Fragment>
      <Navbar />
      <Switch>
        <Route
          exact
          strict
          path={props.match.url}
          render={() => <Redirect to={`${props.match.url}/resources`} />}
        />
        <Route path={`${props.match.url}/resources`} component={ResourceRouting} />
        <Route path={`${props.match.url}/home`} component={Home} />
        <Route path={`${props.match.url}/dashboard`} component={Dashboard} />
        <Route path={`${props.match.url}/patients`} component={Patients} />
        <Route path={`${props.match.url}/analytics`} component={Analytics} />
        <Route component={PageNotFound} />
      </Switch>
    </React.Fragment>
  );
};

const mapStateToProps = (state: ReturnType<typeof rootReducer>) => ({
  loginReducer: state.loginReducer
});

export default connect(mapStateToProps, {})(Layout);
