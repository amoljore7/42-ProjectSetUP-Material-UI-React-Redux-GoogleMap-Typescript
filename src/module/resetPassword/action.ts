import { RESET_PASSWORD } from './types';

export interface userData {
  firstReset: boolean;
  password: string;
}
export interface userPasswordReturnType {
  type: string;
  payload: userData;
}

export const resetPassword: (user: userData) => userPasswordReturnType = (user: userData) => {
  return {
    type: RESET_PASSWORD,
    payload: user
  };
};
