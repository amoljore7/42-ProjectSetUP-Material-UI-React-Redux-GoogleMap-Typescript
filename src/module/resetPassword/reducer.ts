import { RESET_PASSWORD_LOADING, RESET_PASSWORD_SUCCESS, RESET_PASSWORD_FAIL } from './types';

export interface DefaultState {
  loading: boolean;
  error?: string;
}

const defaultState: DefaultState = {
  loading: false,
  error: ''
};

interface actionType {
  type: string;
  payload: string;
}

export const resetPassword = (
  state: DefaultState = defaultState,
  action: actionType
): DefaultState => {
  switch (action.type) {
    case RESET_PASSWORD_LOADING:
      return {
        ...state,
        loading: true
      };
    case RESET_PASSWORD_SUCCESS:
      return {
        ...state,
        loading: false,
        error: ''
      };
    case RESET_PASSWORD_FAIL:
      return {
        ...state,
        loading: false,
        error: action.payload
      };

    default:
      return state;
  }
};
