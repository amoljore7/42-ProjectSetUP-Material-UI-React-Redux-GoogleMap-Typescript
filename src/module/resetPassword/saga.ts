import {
  call,
  CallEffectDescriptor,
  ForkEffect,
  put,
  PutEffect,
  takeEvery,
  SimpleEffect
} from 'redux-saga/effects';
import { AppConfig } from '../../config/app-config';
import Axios, { AxiosResponse } from 'axios';
import {
  RESET_PASSWORD_LOADING,
  RESET_PASSWORD,
  RESET_PASSWORD_SUCCESS,
  RESET_PASSWORD_FAIL
} from './types';

interface payloadData {
  type: string;
  payload: {
    firstReset: boolean;
    password: string;
  };
}

interface resetPasswordResponseType {
  id: string;
  message: string;
  status: string;
  statusCode: string;
}

export function* changeUserPassword(
  payloadData: payloadData
): Generator<
  | PutEffect<{
      type: string;
    }>
  | Promise<resetPasswordResponseType>
> {
  yield put({ type: RESET_PASSWORD_LOADING });
  const id = window.location.href.split('?id=')[1];
  try {
    const response = (yield Axios.put(
      `${AppConfig.serverUrl}/users/updatePassword/${id}`,
      payloadData.payload
    )) as AxiosResponse<resetPasswordResponseType>;

    yield put({ type: RESET_PASSWORD_SUCCESS });

    if (response) {
      window.location.href = '/';
    }
  } catch (error) {
    yield put({ type: RESET_PASSWORD_FAIL, payload: error.message });
  }
}

export function* watchResetPassword(): Generator<ForkEffect<never> | string> {
  yield takeEvery(RESET_PASSWORD, changeUserPassword);
}
