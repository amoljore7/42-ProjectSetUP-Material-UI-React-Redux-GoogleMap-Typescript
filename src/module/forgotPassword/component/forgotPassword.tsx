import React, { useState, useEffect, FormEvent } from 'react';
import { connect } from 'react-redux';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import {
  MuiThemeProvider,
  createMuiTheme,
  Container,
  Button,
  CssBaseline,
  TextField,
  Card,
  CardContent,
  Typography,
  makeStyles,
  CircularProgress,
  Box
} from '@material-ui/core';
import { forgotPassword, userData, userPasswordReturnType } from '../action';
import { DefaultState } from '../../forgotPassword/reducer';
import rootReducer from '../../../root-reducer';
import ErrorString from '../../../utils/error-string';

const useStyles = makeStyles((theme) => ({
  root: {
    '& .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
      borderColor: '#4491E0'
    }
  },
  card: {
    marginTop: theme.spacing(8),
    minWidth: 275,
    padding: '20px'
  },
  title: {
    font: 'normal normal bold 24px/24px Futura',
    marginBottom: '20px',
    color: '#4491E0'
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1)
  },
  signIn: {
    color: '#707070',
    font: 'normal normal bold 26px/35px Futura'
  },
  submitBtn: {
    margin: theme.spacing(3, 0, 2),
    font: 'normal normal medium 14px/16px Roboto'
  },
  circularProgress: {
    marginLeft: 0,
    color: 'white',
    marginRight: '20px'
  },
  headings: {
    font: 'normal normal medium 20px/27px Futura',
    color: '#707070'
  },
  errorMessage: {
    color: 'red'
  }
}));
const themeDark = createMuiTheme({
  palette: {
    background: {
      default: '#F5F5FA'
    }
  }
});

interface StoreProps {
  forgotPasswordReducer: DefaultState;
}

interface DispatchProps {
  forgotPassword: (user: userData) => userPasswordReturnType;
}

type Props = StoreProps & DispatchProps & RouteComponentProps;

export const ForgotPassword: React.FC<Props> = (props: Props) => {
  const classes = useStyles();
  const [submitbtnDisabled, setSubmitBtnDisabled] = useState(false);
  const [formData, setFormData] = useState({
    email: ''
  });

  useEffect(() => {
    setSubmitBtnDisabled(true);
  }, []);

  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;
    setFormData((formData) => ({ ...formData, [name]: value }));
    if (!name.trim() || !value.trim()) {
      setSubmitBtnDisabled(true);
    } else {
      setSubmitBtnDisabled(false);
    }
  };
  const formSubmitHandler = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    props.forgotPassword(formData);
  };

  return (
    <MuiThemeProvider theme={themeDark}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Card className={classes.card} variant="outlined">
          <CardContent>
            <Typography component="h1" variant="h5" className={classes.title}>
              TeleNavigation 2.0
            </Typography>
            <Typography component="h2" variant="h5" className={classes.signIn}>
              Forgot Password
            </Typography>
            <form
              data-testid="formWrapper"
              className={classes.form}
              noValidate
              onSubmit={formSubmitHandler}>
              <h5 className={classes.headings}>
                A Verification link will be sent to your registered email address
              </h5>
              <TextField
                // className={classes.root}
                variant="outlined"
                margin="normal"
                required
                fullWidth
                type="text"
                id="email"
                label="E-mail"
                name="email"
                onChange={inputChangeHandler}
                inputProps={{
                  'data-testid': 'emailInput'
                }}
              />
              <Box>
                <Typography className={classes.errorMessage}>
                  {ErrorString(props.forgotPasswordReducer.error || '') ||
                    props.forgotPasswordReducer.error}
                </Typography>
              </Box>
              <Button
                data-testid="forgotPasswordButton"
                className={classes.submitBtn}
                type="submit"
                // fullWidth
                variant="contained"
                style={{ backgroundColor: '#4491E0', color: '#FFFFFF' }}
                disabled={submitbtnDisabled}>
                {props.forgotPasswordReducer.loading ? (
                  <CircularProgress
                    data-testid="forgotPasswordButtonLoader"
                    className={classes.circularProgress}
                    size={30}
                  />
                ) : (
                  'SEND VERIFICATION EMAIL'
                )}
              </Button>
            </form>
          </CardContent>
        </Card>
      </Container>
    </MuiThemeProvider>
  );
};

const mapStateToProps = (state: ReturnType<typeof rootReducer>) => ({
  forgotPasswordReducer: state.forgotPasswordReducer
});

export default connect(mapStateToProps, { forgotPassword })(withRouter(ForgotPassword));
