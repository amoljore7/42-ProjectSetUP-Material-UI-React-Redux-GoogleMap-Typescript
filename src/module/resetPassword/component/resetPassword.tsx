import React, { useState, useEffect, FormEvent } from 'react';
import { connect } from 'react-redux';
import { withRouter, RouteComponentProps } from 'react-router-dom';
import { notification } from 'antd';
import {
  Container,
  Button,
  CssBaseline,
  TextField,
  Card,
  CardContent,
  Typography,
  makeStyles,
  CircularProgress,
  InputAdornment,
  IconButton,
  Box
} from '@material-ui/core';
import { Visibility, VisibilityOff } from '@material-ui/icons';
import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles';
import Alert from '@material-ui/lab/Alert';

import { resetPassword, userData, userPasswordReturnType } from '../action';
import rootReducer from '../../../root-reducer';
import { DefaultState } from '../../resetPassword/reducer';

const useStyles = makeStyles((theme) => ({
  root: {
    '& .MuiOutlinedInput-root .MuiOutlinedInput-notchedOutline': {
      borderColor: '#4491E0'
    }
  },
  alert: {
    width: '100%',
    '& > * + *': {
      marginTop: theme.spacing(2)
    }
  },
  card: {
    marginTop: theme.spacing(8),
    minWidth: 275,
    padding: '20px'
  },
  title: {
    font: 'normal normal bold 24px/24px Futura',
    marginBottom: '20px',
    color: '#4491E0'
  },
  form: {
    width: '100%', // Fix IE 11 issue.
    marginTop: theme.spacing(1)
  },
  signIn: {
    color: '#707070',
    font: 'normal normal bold 26px/35px Futura'
  },
  submitBtn: {
    margin: theme.spacing(3, 0, 2)
  },
  circularProgress: {
    marginLeft: 0,
    color: 'white',
    marginRight: '20px'
  }
}));
const themeDark = createMuiTheme({
  palette: {
    background: {
      default: '#F5F5FA'
    }
  }
});

interface StoreProps {
  resetPasswordReducer: DefaultState;
}

interface DispatchProps {
  resetPassword: (user: userData) => userPasswordReturnType;
}

type Props = StoreProps & DispatchProps & RouteComponentProps;

export const ResetPassword: React.FC<Props> = (props: Props) => {
  const classes = useStyles();
  const [submitbtnDisabled, setSubmitBtnDisabled] = useState(false);
  const [showPassword_0, setShowPassword_0] = useState(false);
  const [showPassword_1, setShowPassword_1] = useState(false);
  const [formData, setFormData] = useState({
    firstReset: true,
    password: '',
    confirmPassword: ''
  });

  useEffect(() => {
    setSubmitBtnDisabled(true);
  }, []);
  const handleClickShowPassword_1 = () => {
    setShowPassword_1(!showPassword_1);
  };
  const handleClickShowPassword = () => {
    setShowPassword_0(!showPassword_0);
  };

  const inputChangeHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
    const { name, value } = e.target;
    setFormData((formData) => ({ ...formData, [name]: value }));
    if (!formData.password.trim() || !formData.confirmPassword.trim()) {
      setSubmitBtnDisabled(true);
    } else {
      setSubmitBtnDisabled(false);
    }
  };
  const formSubmitHandler = (event: FormEvent<HTMLFormElement>) => {
    event.preventDefault();
    const RE = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[#$@!%&*?])[A-Za-z\d#$@!%&*?]{8,16}$/;
    if (formData.password !== formData.confirmPassword) {
      // alert('Password Not Matching!');
      notification.open({
        message: 'Password Not Matching!',
        duration: 5,
        type: 'error'
      });
      return false;
    } else {
      if (RE.test(formData.password)) {
        props.resetPassword(formData);
      } else {
        notification.open({
          message:
            'Enter Minimum eight characters, at least one uppercase letter, one lowercase letter, one number and one special character',
          duration: 5,
          type: 'error'
        });
      }
    }
  };

  return (
    <MuiThemeProvider theme={themeDark}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Card className={classes.card} variant="outlined">
          <CardContent>
            <Typography component="h1" variant="h5" className={classes.title}>
              TeleNavigation 2.0
            </Typography>
            <Typography component="h2" variant="h5" className={classes.signIn}>
              Reset Password
            </Typography>
            <form
              data-testid="formWrapper"
              className={classes.form}
              noValidate
              onSubmit={formSubmitHandler}>
              <TextField
                // className={classes.root}
                variant="outlined"
                margin="normal"
                required
                fullWidth
                type={showPassword_1 ? 'text' : 'password'}
                id="password"
                label="New Password"
                name="password"
                autoComplete="username"
                onChange={inputChangeHandler}
                inputProps={{
                  'data-testid': 'newPassword'
                }}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <IconButton
                        edge="end"
                        aria-label="toggle password visibility"
                        onClick={handleClickShowPassword_1}>
                        {showPassword_1 ? <VisibilityOff /> : <Visibility />}
                      </IconButton>
                    </InputAdornment>
                  )
                }}
              />
              <TextField
                // className={classes.root}
                variant="outlined"
                margin="normal"
                required
                fullWidth
                name="confirmPassword"
                label="Confirm Password"
                type={showPassword_0 ? 'text' : 'password'}
                id="password"
                autoComplete="current-password"
                onChange={inputChangeHandler}
                inputProps={{
                  'data-testid': 'confirmPassword'
                }}
                InputProps={{
                  endAdornment: (
                    <InputAdornment position="end">
                      <IconButton
                        edge="end"
                        aria-label="toggle password visibility"
                        onClick={handleClickShowPassword}>
                        {showPassword_0 ? <VisibilityOff /> : <Visibility />}
                      </IconButton>
                    </InputAdornment>
                  )
                }}
              />

              <Button
                data-testid="resetPasswordButton"
                className={classes.submitBtn}
                type="submit"
                // fullWidth
                variant="contained"
                style={{ backgroundColor: '#4491E0', color: '#FFFFFF' }}
                disabled={submitbtnDisabled}>
                {props.resetPasswordReducer.loading ? (
                  <CircularProgress
                    data-testid="resetPasswordButtonLoader"
                    className={classes.circularProgress}
                    size={30}
                  />
                ) : (
                  'RESET PASSWORD'
                )}
              </Button>
            </form>
          </CardContent>
        </Card>
      </Container>
    </MuiThemeProvider>
  );
};

const mapStateToProps = (state: ReturnType<typeof rootReducer>) => ({
  resetPasswordReducer: state.resetPasswordReducer
});

export default connect(mapStateToProps, { resetPassword })(withRouter(ResetPassword));
