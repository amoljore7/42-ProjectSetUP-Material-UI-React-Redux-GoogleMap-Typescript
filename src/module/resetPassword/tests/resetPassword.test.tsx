import React from 'react';
import '@testing-library/jest-dom';
import { render, fireEvent, screen, waitFor } from '@testing-library/react';

import { ResetPassword } from '../component/resetPassword';

describe('reset password page', () => {
  it('loads and displays reset password page', async () => {
    const mockProps = {
      history: {} as any,
      location: {} as any,
      match: {} as any,
      resetPasswordReducer: {} as any,
      resetPassword: jest.fn()
    };
    const { asFragment } = render(<ResetPassword {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('formWrapper').children.length).toEqual(3);

    const resetPasswordButton = screen.getByTestId('resetPasswordButton');
    expect(resetPasswordButton).toBeDisabled();

    const newPassword = screen.getByTestId('newPassword');
    const confirmPassword = screen.getByTestId('confirmPassword');

    fireEvent.change(newPassword, { target: { value: 'shubham' } });
    expect(resetPasswordButton).toBeDisabled();

    fireEvent.change(confirmPassword, { target: { value: 'shubham' } });
    waitFor(() => expect(resetPasswordButton).toBeEnabled());
  });
  it('loader is displayed on reset password button when it is clicked', () => {
    const mockProps = {
      history: {} as any,
      location: {} as any,
      match: {} as any,
      resetPasswordReducer: {
        loading: true
      },
      resetPassword: jest.fn()
    };
    render(<ResetPassword {...mockProps} />);
    expect(screen.getByTestId('resetPasswordButtonLoader')).toBeInTheDocument();
  });
  it('loader is not displayed on reset password button when it is not clicked', () => {
    const mockProps = {
      history: {} as any,
      location: {} as any,
      match: {} as any,
      resetPasswordReducer: {
        loading: false
      },
      resetPassword: jest.fn()
    };
    render(<ResetPassword {...mockProps} />);
    expect(screen.getByText('RESET PASSWORD')).toBeInTheDocument();
  });
});
