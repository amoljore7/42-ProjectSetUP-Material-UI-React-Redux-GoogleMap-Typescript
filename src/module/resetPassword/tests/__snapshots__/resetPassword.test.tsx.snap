// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset password page loads and displays reset password page 1`] = `
<DocumentFragment>
  <main
    class="MuiContainer-root MuiContainer-maxWidthXs"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-3 MuiPaper-outlined MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root"
      >
        <h1
          class="MuiTypography-root makeStyles-title-4 MuiTypography-h5"
        >
          TeleNavigation 2.0
        </h1>
        <h2
          class="MuiTypography-root makeStyles-signIn-6 MuiTypography-h5"
        >
          Reset Password
        </h2>
        <form
          class="makeStyles-form-5"
          data-testid="formWrapper"
          novalidate=""
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              New Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                autocomplete="username"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                data-testid="newPassword"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-11"
                >
                  <span>
                    New Password *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              Confirm Password
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                autocomplete="current-password"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                data-testid="confirmPassword"
                id="password"
                name="confirmPassword"
                required=""
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  aria-label="toggle password visibility"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-9 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-11"
                >
                  <span>
                    Confirm Password *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submitBtn-7 Mui-disabled MuiButton-fullWidth Mui-disabled"
            data-testid="resetPasswordButton"
            disabled=""
            style="background-color: rgb(68, 145, 224);"
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              RESET PASSWORD
            </span>
          </button>
        </form>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
